package com.engine.biomine.common;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class FIELDS {

    public static final String id = "id";
    public static final String ji = "ji";
    public static final String ui = "ui";
    public static final String fm = "fm";
    public static final String bm = "bm";

    //------------------ literature
    public static final String abs = "abstract";
    public static final String body = "body";
    public static final String journalTitle = "journalTitle";
    public static final String pmid = "pmid";
    public static final String pmc = "pmc";
    public static final String articleTitle = "title";
    public static final String author = "author";
    public static final String keywords = "keywords";
    public static final String meshTerms = "meshTerms";
//    public static String sections = "sections";
    public static final String referenceTitle = "referenceTitle";
    public static final String referenceId = "referenceId";
    public static final String referenceAuthor = "referenceAuthor";
    public static final String year = "year";
	public static final String captions = "captions";
    //------------------

    //------------------ entity
    public static final String fungus = "fungus";
    public static final String enzyme = "enzyme";
    public static final String accessionNumber = "accessionNumber";
    public static final String glycosation = "glycosation";
    public static final String pH = "pH";
    public static final String productAnalysis = "productAnalysis";
    public static final String temperature = "temperature";
    public static final String entities = "entities";

    //------------------

    public static final String provenance = "provenance";

    //----------------- gff
    public static final String correspondingSeq = "correspondingSeq";
    public static final String featureSource = "featureSource";
    public static final String featureName = "featureName";
    public static final String featureStart = "featureStart";
    public static final String featureEnd = "featureEnd";
    public static final String score = "score";
    public static final String strand = "strand";
    public static final String phase = "phase";
    public static final String attributeId = "attributeId";
    //------------------

    //----------------- fasta
    //public static final String attributeId = "attributeId";
    public static final String geneModelSeq = "geneModelSeq";
    public static final String proteinSeq = "proteinSeq";
    public static final String scaffoldSeq = "scaffoldSeq";
    public static final String genomeName = "genomeName";
    public static final String genomeVersion = "genomeVersion";
    //-----------------

    //----------------- annotation
    public static final String proteinId = "proteinId";
    public static final String proteinLength = "proteinLength";
    public static final String geneName = "geneName";
    public static final String description = "description";
    public static final String evidence = "evidence";
    public static final String evidenceCode = "evidenceCode";
    public static final String cazy = "cazy";
    public static final String cbm = "cbm";
    public static final String gpi = "gpi";
    public static final String kdel = "kdel";
    public static final String wolfPsort = "wolfPsort";
    public static final String spLength = "spLength";
    public static final String goFunction = "goFunction";
    public static final String goProcess = "goProcess";
    public static final String goComponent = "goComponent";
    //-----------------

//    public static List<String> fieldNames = Arrays.asList
//            (id,abs,body,journalTitle,pmc,pmid,articleTitle,
//                    author,keywords,referenceTitle,referenceId,referenceAuthor,
//                    year,captions,proteinID,geneName);
    

}
